app/src/organisms/FirmwareUpdateModal/__tests__/FirmwareUpdateModal.test.tsx

```diff
@@ -95,9 +95,13 @@ describe('FirmwareUpdateModal', () => {
     expect(updateSubsystem).toHaveBeenCalled()
   })
   it('calls refetch instruments and then proceed once update is complete', async () => {
+    jest.useFakeTimers()
     const { getByText } = render(props)
     getByText('A firmware update is required, instrument is updating')
     await waitFor(() => expect(refetch).toHaveBeenCalled())
+    act(() => {
+      jest.advanceTimersByTime(10000)
```

Comment:
just curious why you used 10000 instead of 5000, which is the refetch interval?

```suggestion
      jest.advanceTimersByTime(5000)
```

Reply:
We were seeing a little bit of a delay between when the firmware update was `done` and when the `/instruments` endpoint was returning an `ok` instrument so when we proceeded to the results screen after updating the firmware it always showed the "instrument not attached" error screen, but then the success screen after you hit Try Again. We decided to add a timeout on proceeding to the results, so we keep checking for the `ok` instrument and proceed if we see it, otherwise wait 10000 and then proceed.